src/Symfony/Component/Console/Style/SymfonyStyle.php

```diff
@@ -430,34 +388,6 @@ private function getProgressBar(): ProgressBar
         return $this->progressBar;
     }
 
-    private function autoPrependBlock(): void
-    {
-        $chars = substr(str_replace(\PHP_EOL, "\n", $this->bufferedOutput->fetch()), -2);
-
-        if (!isset($chars[0])) {
-            $this->newLine(); //empty history, so we should start with a new line.
-
-            return;
-        }
-        //Prepend new line for each non LF chars (This means no blank line was output before)
-        $this->newLine(2 - substr_count($chars, "\n"));
-    }
-
-    private function autoPrependText(): void
-    {
-        $fetched = $this->bufferedOutput->fetch();
-        //Prepend new line if last char isn't EOL:
-        if ("\n" !== substr($fetched, -1)) {
-            $this->newLine();
-        }
-    }
-
-    private function writeBuffer(string $message, bool $newLine, int $type): void
-    {
-        // We need to know if the last chars are PHP_EOL
-        $this->bufferedOutput->write($message, $newLine, $type);
-    }
-
     private function createBlock(iterable $messages, string $type = null, string $style = null, string $prefix = ' ', bool $padding = false, bool $escape = false): array
```

Comment:
I think this method should also make its way in the new class.

Reply:
Totally, thanks. I've extracted this to the new OutputBlock class